From 5796b079cc4ff4ea685745c86dfd07135b7cadc6 Mon Sep 17 00:00:00 2001
From: Dirk Mueller <dirk@dmllr.de>
Date: Fri, 12 Jun 2020 19:20:09 +0200
Subject: [PATCH 08/10] helm: allow to configure bpf-nat-global-max using Helm

Set the value to the current value of option.NATMapEntriesGlobalDefault

A successive PR will reduce it for #10056

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>

Cherry-Pick of https://github.com/cilium/cilium/commit/20f6083d6fabfcce302a1c43d81ddf639a23f7a6
---
 .../kubernetes/cilium/charts/config/templates/configmap.yaml  | 4 ++++
 install/kubernetes/cilium/values.yaml                         | 3 +++
 install/kubernetes/quick-install.yaml                         | 4 ++++
 3 files changed, 11 insertions(+)

diff --git a/install/kubernetes/cilium/charts/config/templates/configmap.yaml b/install/kubernetes/cilium/charts/config/templates/configmap.yaml
index ebc2a0158..7bcc40b23 100644
--- a/install/kubernetes/cilium/charts/config/templates/configmap.yaml
+++ b/install/kubernetes/cilium/charts/config/templates/configmap.yaml
@@ -136,6 +136,10 @@ data:
   # policy map (per endpoint)
   bpf-policy-map-max: "{{ .Values.global.bpf.policyMapMax }}"
 
+  # bpf-nat-global-max specified the maximum number of entries in the BPF NAT
+  # table.
+  bpf-nat-global-max: "{{ .Values.global.bpf.natMax }}"
+
   # Pre-allocation of map entries allows per-packet latency to be reduced, at
   # the expense of up-front memory allocation for the entries in the maps. The
   # default value below will minimize memory usage in the default installation;
diff --git a/install/kubernetes/cilium/values.yaml b/install/kubernetes/cilium/values.yaml
index 138375932..92e4d3c1b 100644
--- a/install/kubernetes/cilium/values.yaml
+++ b/install/kubernetes/cilium/values.yaml
@@ -213,6 +213,9 @@ global:
     # tracking table
     ctAnyMax: 262144
 
+    # natMax is the maximum number of entries for the NAT table
+    natMax: 841429
+
     # policyMapMax is the maximum number of entries in endpoint policy map (per endpoint)
     policyMapMax: 16384
 
diff --git a/install/kubernetes/quick-install.yaml b/install/kubernetes/quick-install.yaml
index 4280c3377..bc106477b 100644
--- a/install/kubernetes/quick-install.yaml
+++ b/install/kubernetes/quick-install.yaml
@@ -79,6 +79,10 @@ data:
   # policy map (per endpoint)
   bpf-policy-map-max: "16384"
 
+  # bpf-nat-global-max specified the maximum number of entries in the BPF NAT
+  # table.
+  bpf-nat-global-max: "841429"
+
   # Pre-allocation of map entries allows per-packet latency to be reduced, at
   # the expense of up-front memory allocation for the entries in the maps. The
   # default value below will minimize memory usage in the default installation;
-- 
2.27.0

